python/graphscope/framework/graph.py

```diff
@@ -1068,49 +1056,102 @@ def _check_unmodified(self):
+            types_pb2.VID_TYPE: utils.s_to_attr("uint64_t"),
+            types_pb2.IS_FROM_VINEYARD_ID: utils.b_to_attr(False),
+            types_pb2.IS_FROM_GAR: utils.b_to_attr(True),
+            types_pb2.VERTEX_MAP_TYPE: utils.i_to_attr(vertex_map),
+            types_pb2.COMPACT_EDGES: utils.b_to_attr(False),
+            types_pb2.GRAPH_INFO_PATH: utils.s_to_attr(path),
+        }
+        op = dag_utils.create_graph(
+            sess.session_id, graph_def_pb2.ARROW_PROPERTY, inputs=[], attrs=config
+        )
+        return sess._wrapper(GraphDAGNode(sess, op))
 
     @classmethod
-    def load_from(cls, path, sess, **kwargs):
-        """Construct a `Graph` by deserialize from `path`.
-        It will read all serialization files, which is dumped by
-        `Graph.serialize`.
-        If any serialize file doesn't exists or broken, will error out.
+    def load_from(cls, uristring, sess=None):
+        """Load a ArrowProperty graph from with a certain data source. The data source
+        can be vineyard serialized files or graphar files.
 
         Args:
-            path (str): Path contains the serialization files.
+            uristring (str): URI contains the description of the data source.
+            example: "graphar+file:///tmp/graphar/xxx"
             sess (`graphscope.Session`): The target session
                 that the graph will be construct in
-
         Returns:
             `Graph`: A new graph object. Schema and data is supposed to be
-                identical with the one that called serialized method.
         """
-        op = dag_utils.load_graph_from(path, sess, **kwargs)
-        return sess._wrapper(GraphDAGNode(sess, op))
+        from graphscope.client.session import get_default_session
+
+        if sess is None:
+            sess = get_default_session()
+        uri = urlparse(uristring)
+        if uri.scheme and "+" in uri.scheme:
+            source = uri.scheme.split("+")[0]
+            path = uri.scheme.split("+")[-1] + "://" + uri.netloc + uri.path
+            if source == "graphar":
+                return cls._load_from_graphar(path, sess)
+            else:
+                raise ValueError("Unknown source: %s" % source)
+        else:
+            # not a uri string, assume it is a path for deserialization
+            op = dag_utils.deserialize_graph(uristring, sess)
+            return sess._wrapper(GraphDAGNode(sess, op))
 
-    def archive(self, path):
-        """Archive graph gar format files base on the graph info.
-        The meta and data of graph is dumped to specified location,
-        and can be restored by `Graph.deserialize` in other sessions.
+    def save_to(
+        self,
+        path,
+        format="serialization",
+        graphar_graph_name="graph",
+        graphar_vertex_block_size=262144,  # 2^18
+        graphar_edge_block_size=4194304,  # 2^22
+        graphar_file_format="parquet",
+        graphar_version="v1",
```

Comment:
I think it's more clear to pass graphar related parameters in a dict, not messed up with other parameters.

Reply:
that's our design discuss in issue #2957 , and the parameters are like `Session` __init__ method's parameter.